src/Search.php

```diff
@@ -1750,10 +1748,16 @@ public static function displayData(array $data)
             'may_be_browsed'      => Toolbox::hasTrait($item, \Glpi\Features\TreeBrowse::class),
         ]);
 
-       // Add items in item list
+        // Add items in item list
         foreach ($data['data']['rows'] as $row) {
-            $current_type = (isset($row['TYPE']) ? $row['TYPE'] : $data['itemtype']);
-            Session::addToNavigateListItems($current_type, $row["id"]);
+            if ($itemtype !== AllAssets::class) {
+                Session::addToNavigateListItems($itemtype, $row["id"]);
+            } else {
+                // In case of a global search, reset and empty navigation list to ensure navigation in
+                // item header context is not shown. Indeed, this list does not support navigation through
+                // multiple itemtypes, so it should not be displayed in global search context.
+                Session::initNavigateListItems($row['TYPE'] ?? $data['itemtype']);
```

Comment:
i don't understand how this code reset the navigation list (and so, do not display navigation header).
Do you mind explaining a little bit ?

Reply:
On line 1716, call to `Session::initNavigateListItems($data['itemtype']);` will init/reset the list for current itemtype (i.e. `AllAssets` which is not used actually).
On this loop, we call this method on current row itemtype, in order to reset any existing list in session.

For instance, if a user did a search on computers earlier in its session, `$_SESSION['glpilistitems']['Computer']` will contains some elements. With previous code, elements from global search were added to existing elements in session, so list was neither matching the previous Computer search. If user did a second global search, new elements were added and list was growing again. So when user was going to any element of its global search, navigation was refering to some elements from a previous search combined with elements from global search matching the displayed item type.

With proposed code, on global view context, list of itemtypes of all matching elements are reset (via the `Session::initNavigateListItems()` method). When user displays an item from its global search, item list in session is empty, so navigation through this list is not proposed.